From 7f3a28b8c03896e850aef8c6f35e6f6ba8bd7076 Mon Sep 17 00:00:00 2001
From: Russell King <rmk+kernel@arm.linux.org.uk>
Date: Sun, 22 Nov 2015 23:01:45 +0000
Subject: [PATCH 169/194] staging: etnaviv: remove etnaviv object mm_list

We now have no use to maintain a list of active GPU objects, so remove
the mm_list node from the etnaviv objects, and its initialisers, users
and the GPU active list.

Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
---
 drivers/staging/etnaviv/etnaviv_gem.c | 5 -----
 drivers/staging/etnaviv/etnaviv_gem.h | 1 -
 drivers/staging/etnaviv/etnaviv_gpu.c | 3 ---
 drivers/staging/etnaviv/etnaviv_gpu.h | 3 ---
 4 files changed, 12 deletions(-)

diff --git a/drivers/staging/etnaviv/etnaviv_gem.c b/drivers/staging/etnaviv/etnaviv_gem.c
index 392b12e..a239754 100644
--- a/drivers/staging/etnaviv/etnaviv_gem.c
+++ b/drivers/staging/etnaviv/etnaviv_gem.c
@@ -335,9 +335,6 @@ void etnaviv_gem_move_to_active(struct drm_gem_object *obj,
 		etnaviv_obj->write_fence = fence;
 
 	etnaviv_obj->access |= access;
-
-	list_del_init(&etnaviv_obj->mm_list);
-	list_add_tail(&etnaviv_obj->mm_list, &gpu->active_list);
 }
 
 void etnaviv_gem_move_to_inactive(struct drm_gem_object *obj)
@@ -351,7 +348,6 @@ void etnaviv_gem_move_to_inactive(struct drm_gem_object *obj)
 	etnaviv_obj->read_fence = 0;
 	etnaviv_obj->write_fence = 0;
 	etnaviv_obj->access = 0;
-	list_del_init(&etnaviv_obj->mm_list);
 }
 
 static inline enum dma_data_direction etnaviv_op_to_dma_dir(u32 op)
@@ -561,7 +557,6 @@ static int etnaviv_gem_new_impl(struct drm_device *dev, u32 size, u32 flags,
 		reservation_object_init(&etnaviv_obj->_resv);
 	}
 
-	INIT_LIST_HEAD(&etnaviv_obj->mm_list);
 	INIT_LIST_HEAD(&etnaviv_obj->vram_list);
 
 	*obj = &etnaviv_obj->base;
diff --git a/drivers/staging/etnaviv/etnaviv_gem.h b/drivers/staging/etnaviv/etnaviv_gem.h
index 45136d1..8ad9b31 100644
--- a/drivers/staging/etnaviv/etnaviv_gem.h
+++ b/drivers/staging/etnaviv/etnaviv_gem.h
@@ -48,7 +48,6 @@ struct etnaviv_gem_object {
 	u32 flags;
 
 	struct list_head gem_node;
-	struct list_head mm_list;
 	struct etnaviv_gpu *gpu;     /* non-null if active */
 	u32 access;
 	u32 read_fence, write_fence;
diff --git a/drivers/staging/etnaviv/etnaviv_gpu.c b/drivers/staging/etnaviv/etnaviv_gpu.c
index 23d6aaa..89fdb47 100644
--- a/drivers/staging/etnaviv/etnaviv_gpu.c
+++ b/drivers/staging/etnaviv/etnaviv_gpu.c
@@ -1318,7 +1318,6 @@ static int etnaviv_gpu_bind(struct device *dev, struct device *master,
 
 	gpu->drm = drm;
 
-	INIT_LIST_HEAD(&gpu->active_list);
 	INIT_LIST_HEAD(&gpu->active_cmd_list);
 	INIT_WORK(&gpu->retire_work, retire_worker);
 	INIT_WORK(&gpu->recover_work, recover_worker);
@@ -1344,8 +1343,6 @@ static void etnaviv_gpu_unbind(struct device *dev, struct device *master,
 
 	hangcheck_disable(gpu);
 
-	WARN_ON(!list_empty(&gpu->active_list));
-
 #ifdef CONFIG_PM
 	pm_runtime_get_sync(gpu->dev);
 	pm_runtime_put_sync_suspend(gpu->dev);
diff --git a/drivers/staging/etnaviv/etnaviv_gpu.h b/drivers/staging/etnaviv/etnaviv_gpu.h
index c1c849f..df40c94 100644
--- a/drivers/staging/etnaviv/etnaviv_gpu.h
+++ b/drivers/staging/etnaviv/etnaviv_gpu.h
@@ -102,9 +102,6 @@ struct etnaviv_gpu {
 	struct completion event_free;
 	spinlock_t event_spinlock;
 
-	/* list of GEM active objects: */
-	struct list_head active_list;
-
 	/* list of currently in-flight command buffers */
 	struct list_head active_cmd_list;
 
-- 
2.6.2

